### PR TITLE
Eventhub track1 patch for close on checkpoint failure

### PR DESCRIPTION
**Pull request by KieranBrantnerMagee**
Add configurable short circuiting to the event processor host via the _release_partition_on_checkpoint_failure eph option, to cause the eph to proactively release its partition on checkpoint persist failure, rather than continuing to try and consume off it.

This will mitigate the number of collisions/multiple reads, with the potential impact of minor latency in the case that the underlying interruption was transient and the partition must be retaken.

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

**Comment by lmazuel** [comment]
@KieranBrantnerMagee what is the context of this change? Customer issue?
Why the parameter starts by an underscore? What's the documentation story about it.

**Comment by KieranBrantnerMagee** [comment]
> @KieranBrantnerMagee what is the context of this change? Customer issue?
> Why the parameter starts by an underscore? What's the documentation story about it.

In hindsight I should have echoed our OOB discussion here :)  For posterity: Customer issue.  In situations with a number of distributed consumers, there can be situations in which a partition is stolen with the other consumer being somewhat leisurely in its decision to relinquish that partition.  This change simply makes it eager to drop the lease on failure and fetch a new one.  This heavily reduces message re-processing and partition ownership contention.

The param started with an underscore because I was hesitant to make a slightly esoteric feature implied as supported/in mainline, but given your point that this is a pre 1.0, I'll be taking your feedback to make it public.

Documentation story thus far is just in the local docstring, alongside this fix I'll see if I can find any other appropriate locations to mention this in the broader docs/sync with our docs folks about it.

File: sdk/eventhub/azure-eventhubs/azure/eventhub/__init__.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
-__version__ = "1.3.2"
+__version__ = "1.3.3b1"
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhubs/azure/eventhub/__init__.py]: I'm guessing we should revert these version changes in favour of those made in the other PR?

Comment (KieranBrantnerMagee) [reply sdk/eventhub/azure-eventhubs/azure/eventhub/__init__.py]: Apt timing, I was about to start a discussion with you about how we want to handle this going forward.  I'm open to any preferences you have one way or the other, as you see I was somewhat ambivalent as to which we settled on :P

**Pull request merged**